application/src/main/java/org/togetherjava/tjbot/commands/moderation/WhoIsCommand.java

```diff
@@ -0,0 +1,263 @@
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+
+import javax.annotation.CheckReturnValue;
+import java.awt.*;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * You ever wanted to stalk someone on Discord? Well no need to worry anymore! I got you covered,
+ * this command allows you to look up user (or member) info.
+ */
+@SuppressWarnings("ClassWithoutLogger")
+public final class WhoIsCommand extends SlashCommandAdapter {
+    private static final String USER_OPTION = "user";
+    private static final String SHOW_SERVER_SPECIFIC_INFO = "show_server_specific_info";
+
+    // Sun, December 11, 2016, 13:36:30
+    private static final DateTimeFormatter DATE_TIME_FORMAT =
+            DateTimeFormatter.ofPattern("E, MMMM d, u, HH:mm:ss");
+
+    /**
+     * Creates an instance of who is command
+     */
+    public WhoIsCommand() {
+        super("whois", "Provides info about the given user", SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, USER_OPTION, "the user to look up", true)
+            .addOption(OptionType.BOOLEAN, SHOW_SERVER_SPECIFIC_INFO,
+                    "whenever to show info that is specific to this server, such as ", false);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        OptionMapping userOption = Objects.requireNonNull(event.getOption(USER_OPTION),
+                "The given user option cannot be null");
+        OptionMapping showServerSpecificInfoOption = event.getOption(SHOW_SERVER_SPECIFIC_INFO);
+
+        User user = userOption.getAsUser();
+        Member member = userOption.getAsMember();
+
+
+        boolean showServerSpecificInfo = null != member && (null == showServerSpecificInfoOption
+                || showServerSpecificInfoOption.getAsBoolean());
+
+        user.retrieveProfile().flatMap(profile -> {
+            if (showServerSpecificInfo) {
+                return handleWhoIsMember(event, member, profile);
+            } else {
+                return handleWhoIsUser(event, user, profile);
+            }
+        }).queue();
+    }
+
+    @CheckReturnValue
+    private static @NotNull ReplyAction handleWhoIsUser(final @NotNull SlashCommandEvent event,
+            final @NotNull User user, final @NotNull User.Profile profile) {
+
+        StringBuilder descriptionBuilder =
+                new StringBuilder().append(handleUserIdentification(user))
+                    .append("\n**Is bot:** ")
+                    .append(user.isBot())
+                    .append(handleUserFlags(user.getFlags()))
+                    .append("\n**Registration date:** ")
+                    .append(DATE_TIME_FORMAT.format(user.getTimeCreated()));
```

Comment:
NIT: For such small strings, string concat is totally okay as well. Elaborate if it would be more readable to change it to a formatted triple-string like:
```java
"""
foo: %s
bar: %s
baz: %s
""".formatted(foo, bar, baz)
```

Reply:
I think the triple strings might be better here.

Reply:
So
```java
"""
**Is bot:** %s
%s
**Registration date:** %s
""".formatted(user.isBot(), handleUserFlags(user.getFlags()), DATE_TIME_FORMAT.format(user.getTimeCreated());
```

This has no benefit to readability, it even makes it worse due to the random `%s` imho

Reply:
Really? For me its easier to read cause I can focus on the structure of the message.
With your original code its easier to detect whether the correct args are given but the message itself is less obvious to read.

Reply:
I personally don't like it because it adds these `%s` without indication of what they are.
So this means you've to read the arguments, the second argument to be specific, handleUserflags?
Current code clearly shows what happens I think personally.

To show what I mean with your example, the perfect world:
```java
"""
foo: %s
bar: %s
baz: %s
""".formatted(foo, bar, baz)
``` 
Reality:
```java
"""
foo: %s
%s
baz: %s
""".formatted(foo, bar, baz)
```
has a weird %s, and you don't know what it is, you've to read the arguments to figure out what one it is.
So first you've to count the amount of `%s`, then you've to find where we format, and go to that `...` argument.
To read the name/method/whatever, to understand the message structure

Reply:
Thats fair. Its a pitty that there are no "named placeholders" or similar. Feel free to resolve this conversation.

Reply:
If they existed, I would've been using them!